domain/model/state/modelstate.go

```diff
@@ -219,6 +219,38 @@ func (s *ModelState) GetModelMetrics(ctx context.Context) (coremodel.ModelMetric
 	}, nil
 }
 
+// GetModelCloudType returns the cloud type from a read-only model that has been
+// set in the database. If no model exists then an error satisfying
+// [modelerrors.NotFound] is returned.
+func (s *ModelState) GetModelCloudType(ctx context.Context) (string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	m := dbReadOnlyModel{}
+	stmt, err := s.Prepare(`SELECT &dbReadOnlyModel.cloud_type FROM model`, m)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt).Get(&m)
+		if errors.Is(err, sql.ErrNoRows) {
+			return errors.New("model does not exist").Add(modelerrors.NotFound)
+		}
+		return err
+	})
+
+	if err != nil {
+		return "", errors.Errorf(
+			"getting cloud type from model state: %w", err,
+		)
+	}
```

Comment:
We don't need the annotation here. It adds no value - what we care about knowing are the errors from inside the txn closure which have been annotated. The caller of this method will annotate if needed, and if it does, we'll end up with stuttering `getting cloud type blah blah: getting cloud type ...`

Reply:
It's not clear what you want here. Are you able to provide some concrete direction here with an example and also look at the current call site from the service to make sure that is inline with your expectations.

Cheers.

Reply:
Thanks for the feedback, Tom and Ian! I see where you're coming from, Tom, in terms of ensuring clarity at the call site. However, I agree with Ian's point that the annotation for transaction errors might not be necessary here, and it could also result in redundancy if the caller method already handles annotation. Perhaps I could shift this error annotation to GetEnvironVersion() in ModelService so that we can better have clarity. Happy to discuss further if you think this approach needs tweaking!